cdap-app-fabric/src/main/java/co/cask/cdap/app/store/Store.java

```diff
@@ -341,7 +341,7 @@ WorkflowStatistics getWorkflowStatistics(Id.Workflow workflowId, long startTime,
    * @param timeInterval The timeInterval with which the user wants to space out the runs
    * @return Map of runId of Workflow to DetailedStatistics of the run
    */
-  Collection<WorkflowDataset.WorkflowRunRecord> retrieveSpacedRecords(Id.Workflow workflow, String runId,
+  Collection<WorkflowDataset.WorkflowRunRecord> retrieveSpacedRecords(ProgramId workflow, String runId,
```

Comment:
@maochf, let's add a ``WorkflowId`` class, since some functionality is specific to workflows only.